daemon/exec_runner.go

```diff
@@ -117,7 +117,15 @@ func (r *ExecRunner) ResetRestartCounter() {
 }
 
 func (r *ExecRunner) ValidateBeforeStart() error {
-	_, err := exec.LookPath(r.exec)
+	whitelisted, err := common.PathMatch(r.GetBackend().ExecutablePath, r.context.UserConfig.CollectorBinariesWhitelist)
+	if err != nil {
+		return errors.New("Can not validate binary path")
+	}
+	if !whitelisted && len(r.context.UserConfig.CollectorBinariesWhitelist) > 0 {
+		return r.backend.SetStatusLogErrorf("Can not execute collector %s, binary is not whitelisted", r.backend.ExecutablePath)
```

Comment:
Please include the config option name to make it really clear where to find the option. Something like:

```
Couldn't execute collector %s, binary path is not included in `collector_binaries_whitelist' config option.
```